test/crawler_web/controllers/home_controller_test.exs

```diff
@@ -23,4 +26,72 @@ defmodule CrawlerWeb.HomeControllerTest do
       assert html_response(conn, 200) =~ "#{gettext("Log in")}</a>"
     end
   end
+
+  describe "POST /home" do
+    test "given valid keyword csv file, creates keyword successfully", %{conn: conn} do
+      user = insert(:user)
+      uploaded_file = keyword_file_fixture("valid.csv")
+
+      conn =
+        conn
+        |> log_in_user(user)
+        |> post(Routes.home_path(conn, :create), %{keyword_csv_file: %{file: uploaded_file}})
+
+      assert get_flash(conn, :info) == gettext("Keywords were uploaded!")
+    end
+
+    test "given empty file, shows validation error", %{conn: conn} do
+      user = insert(:user)
+      upload_file = keyword_file_fixture("empty.csv")
+
+      conn =
+        conn
+        |> log_in_user(user)
+        |> post(Routes.home_path(conn, :create), %{keyword_csv_file: %{file: upload_file}})
+
+      assert get_flash(conn, :info) == nil
+      assert get_flash(conn, :error) == "The file is empty."
+    end
+
+    test "given big file, shows validation error", %{conn: conn} do
+      user = insert(:user)
+      upload_file = keyword_file_fixture("big.csv")
+
+      conn =
+        conn
+        |> log_in_user(user)
+        |> post(Routes.home_path(conn, :create), %{keyword_csv_file: %{file: upload_file}})
+
+      assert get_flash(conn, :info) == nil
+      assert get_flash(conn, :error) == "The file is too big, allowed size is up to 1000 keywords."
+    end
+
+    test "given non CSV file, shows validation error", %{conn: conn} do
+      user = insert(:user)
+      upload_file = keyword_file_fixture("non_csv.txt")
+
+      conn =
+        conn
+        |> log_in_user(user)
+        |> post(Routes.home_path(conn, :create), %{keyword_csv_file: %{file: upload_file}})
+
+      assert get_flash(conn, :info) == nil
+      assert get_flash(conn, :error) == "The file doesn't look like CSV."
```

Comment:
Should we keep it the same with other error message which doesn't have the full stop :D 
```suggestion
      assert get_flash(conn, :error) == "The file doesn't look like CSV"
```